mne/decoding/classifier.py

```diff
@@ -453,3 +453,24 @@ def transform(self, epochs_data, y=None):
                                      picks=self.picks, n_jobs=self.n_jobs,
                                      copy=False, verbose=False)
         return epochs_data
+
+
+def compute_patterns(epochs, linear_model):
+    from sklearn.preprocessing import StandardScaler
+    labels = epochs.events[:, -1]
+    evoked = epochs.average()
+    # concatenate channel x time data
+    X = epochs.get_data().reshape(len(epochs), -1)
+    # normalize the data
+    sc = StandardScaler()
+    X = sc.fit_transform(X)
+    # fit the linear model
+    linear_model.fit(X, labels)
+    # predict the train data
+    y_pred = linear_model.predict(X)
+    # computes the patterns and reshape it
+    patterns = np.dot(y_pred, X).reshape(len(epochs.ch_names), 
+                                         len(epochs.times))
```

Comment:
Formula 7 states:

    A = dot(cov(X), W) = cov(X, y_pred)

In your code, a simple dot multiplication is performed. I'm assuming this boils down to the emperical estimate of the covariance. It would be nice to support some [shrinkage](http://scikit-learn.org/stable/modules/covariance.html) in there as well.

Why not do the OLS in all cases? Formula 6 is just the special K=1 case. When treating the problem as a regression problem you gain a lot of benefits (as pointed out in the paper), such as straightforward ways to incorporate shrinkage and other priors. This approach should gracefully degrade to the computation you perform now when K=1.